modules/apps/27-interchain-accounts/keeper/relay.go

```diff
@@ -75,60 +75,46 @@ func (k Keeper) createOutgoingPacket(
 	return packet.Sequence, nil
 }
 
-func (k Keeper) AuthenticateTx(ctx sdk.Context, msgs []sdk.Msg, portId string) error {
-	seen := map[string]bool{}
-	var signers []sdk.AccAddress
-	for _, msg := range msgs {
-		for _, addr := range msg.GetSigners() {
-			if !seen[addr.String()] {
-				signers = append(signers, addr)
-				seen[addr.String()] = true
-			}
-		}
-	}
-
-	interchainAccountAddr, found := k.GetInterchainAccountAddress(ctx, portId)
+// AuthenticateTx ensures the provided msgs contain the correct interchain account signer address retrieved
+// from state using the provided controller port identifier
+func (k Keeper) AuthenticateTx(ctx sdk.Context, msgs []sdk.Msg, portID string) error {
+	interchainAccountAddr, found := k.GetInterchainAccountAddress(ctx, portID)
 	if !found {
-		return sdkerrors.ErrUnauthorized
+		return sdkerrors.Wrapf(types.ErrInterchainAccountNotFound, "failed to retrieve interchain account on port %s", portID)
 	}
 
-	for _, signer := range signers {
-		if interchainAccountAddr != signer.String() {
-			return sdkerrors.ErrUnauthorized
+	for _, msg := range msgs {
+		for _, signer := range msg.GetSigners() {
+			if interchainAccountAddr != signer.String() {
+				return sdkerrors.Wrap(sdkerrors.ErrUnauthorized, "failed to authenticate interchain account transaction")
```

Comment:
should we print the expected signer?

Reply:
My instinct is no as it could potentially allow a malicious party to find out what the associated address they're trying to access is. 

Maybe that is overly paranoid though?

Reply:
This information is in the state machine, so it is public already

Reply:
Yeah, I can't think of anything either, but unless there is a clear benefit to logging it out I usually air on the side of caution for these things. Is there any reason why knowing the associated address would be helpful?

Just an instinctual feeling :thinking:

Reply:
It seems fine to me to include it. If anything it will help debug an issue if it comes to it. But in reality something has gone fundamentally wrong if we ever reach this error, in terms of handshake flow and data stored in state. 

I'll update to the following:
```go
return sdkerrors.Wrapf(sdkerrors.ErrUnauthorized, "unexpected signer address: expected %s, got %s", interchainAccountAddr, signer.String())
```

Reply:
Fine with me then :+1: 

> But in reality something has gone fundamentally wrong if we ever reach this error
Won't this error be reached if a malicious controller chain tries to put a signer address that isn't an interchain account they own?

Reply:
Yeah, good point! 💯